### DISCUSSION TITLE
Constructing the `sample_pareto_frontiers` for the `qMultiObjectiveMaxValueEntropy` acquisition function

### DISCUSSION THREAD

**Discussion by mamunm** [meta-pytorch/botorch#2405] [Q&A] [answered]
In the `botorch.acquisition.multi_objective.max_value_entropy_search.qMultiObjectiveMaxValueEntropy` function, it requires a `sample_pareto_frontiers` callable, which will be used to construct the num_samples sampled Pareto frontier (link: https://botorch.org/api/_modules/botorch/acquisition/multi_objective/max_value_entropy_search.html#qMultiObjectiveMaxValueEntropy). I couldn't find an example of how to construct it. How do we build the sample_pareto_frontiers for a multiobjective optimization problem?

I am thinking of something like this:

```
def sample_pareto_frontier(X):
    pareto_front = is_non_dominated(X)
    return X[pareto_front]
```
But it doesn't look right as the output should be `num_samples * n * m`.

**Comment by esantorella** [comment, marked as answer]
Hi, I *think* the following should work, but let me know if you run into issues with it:


```python
from botorch.acquisition.multi_objective.utils import sample_optimal_points

def sample_pareto_frontier(model):
    pareto_sets, pareto_frontiers = sample_optimal_points(
        model, bounds=bounds, num_samples=num_samples, num_points=num_points
    )
    return pareto_fronts
```

You would need to define `bounds`, `num_samples`, and `num_points`.
